### PR TITLE
PixelPaint: Add Fastboxblur and Bloom Filters

### PR DESCRIPTION
**Pull request by TobyAsE**
This PR adds two new Filters to PixelPaint:
- FastBoxBlur: This filter lived inside LibGfx for a long time, now exposed to PixelPaint
image
To get this filter to work, I extended the FBBF to also know what to do with the ImageFormat BGRx8888.

- Bloom: An artistic filter for mimicking bleeding of bright areas into dark ones in real cameras
image
For this filter, I added a LumaFilter to LibGfx, that lets you keep parts of an image that lie within a certain luminosity threshold. And  also a BitmapMixer I use to draw two Bitmaps onto each other, this is very naive but gets the job done for now.
This is what it looks like: left before, right after (done entirely in PixelPaint):
bloom-example

### CODE REVIEW

File: Userland/Libraries/LibGfx/Filters/LumaFilter.cpp [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+void LumaFilter::apply(u8 lower_bound, u8 upper_bound)
+{
+    if (upper_bound < lower_bound)
+        return;
+
+    int height = m_bitmap.height();
+    int width = m_bitmap.width();
+
+    auto format = m_bitmap.format();
+    VERIFY(format == BitmapFormat::BGRA8888 || format == BitmapFormat::BGRx8888);
```
Comment (IdanHo) [inline Userland/Libraries/LibGfx/Filters/LumaFilter.cpp:19]: Is there a specific reason for verifying this here? It doesn't look like the filter assumes it.

File: Userland/Libraries/LibGfx/Filters/LumaFilter.cpp [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+    for (int y = 0; y < height; ++y) {
+        for (int x = 0; x < width; ++x) {
+            Color color;
+            color = m_bitmap.get_pixel(x, y);
+
+            auto luma = color.luminosity();
+            if (lower_bound > luma || upper_bound < luma) {
+                m_bitmap.set_pixel(x, y, { 0, 0, 0, color.alpha() });
```
Comment (IdanHo) [inline Userland/Libraries/LibGfx/Filters/LumaFilter.cpp:28]: no need for the curlies `{}`

File: Userland/Libraries/LibGfx/Filters/LumaFilter.cpp [hunk 3]
```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (c) 2022, Tobias Christiansen <tobyase@serenityos.org>
```
Comment (IdanHo) [inline Userland/Libraries/LibGfx/Filters/LumaFilter.cpp:2]: This commit message looks to be missing a word as well?

File: Userland/Applications/PixelPaint/Filters/Bloom.cpp [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+
+        auto& radius_slider = radius_container.add<GUI::ValueSlider>(Orientation::Horizontal, "px");
+        radius_slider.set_range(0, 50);
+        radius_slider.set_value(m_blur_radius);
+        radius_slider.on_change = [&](int value) {
+            m_blur_radius = value;
+        };
+    }
+
+    return m_settings_widget.ptr();
```
Comment (IdanHo) [inline Userland/Applications/PixelPaint/Filters/Bloom.cpp:84]: useless `.ptr()`

**Review by IdanHo — COMMENTED** [review]
Nice work, but a couple of nitpicks:

File: Userland/Libraries/LibGfx/Filters/LumaFilter.cpp (hunk 1 above)
Comment (TobyAsE) [reply Userland/Libraries/LibGfx/Filters/LumaFilter.cpp:19]: It was there before, I only extended it.
The value is used to determine which of the {get,set}_pixel() functions to use.

Those two Formats are afaict the ones passed around by PixelPaint.
We only support both Formats and not the other ones. I don't know enough about what is going on there to also support the others. If anybody needs other formats, they will hit the VERIFY and have to implement it themselves.

Comment (IdanHo) [reply Userland/Libraries/LibGfx/Filters/LumaFilter.cpp:19]: Sure, that sounds reasonable.

File: Userland/Libraries/LibGfx/BitmapMixer.cpp [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+                break;
+            case MixingMethod::Lightest:
+                auto original_lightness = original_color.red() + original_color.green() + original_color.blue();
+                auto other_lightness = other_color.red() + other_color.green() + other_color.blue();
+                if (original_lightness > other_lightness) {
+                    output_color = original_color;
+                } else {
+                    output_color.set_red(other_color.red());
+                    output_color.set_green(other_color.green());
+                    output_color.set_blue(other_color.blue());
```
[the remark this answers is not in the document]
Comment (TobyAsE) [reply Userland/Libraries/LibGfx/BitmapMixer.cpp:38]: I want to keep the alpha of `original_color` set in ln 23.
I could use `output_color = other_color` and then set its alpha to original_color's alpha but that's essentially the same.

[the remark this answers is not in the document]
Comment (IdanHo) [reply Userland/Libraries/LibGfx/BitmapMixer.cpp:38]: Sorry, forgot about the alpha!

File: Userland/Libraries/LibGfx/BitmapMixer.cpp [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+                }
+                break;
+            }
+            if (original_color != output_color)
+                m_bitmap.set_pixel(x, y, output_color);
```
[the remark this answers is not in the document]
Comment (TobyAsE) [reply Userland/Libraries/LibGfx/BitmapMixer.cpp:43]: Do you have an idea for a better name?

[the remark this answers is not in the document]
Comment (IdanHo) [reply Userland/Libraries/LibGfx/BitmapMixer.cpp:43]: `mix_with`?

[the remark this answers is not in the document]
Comment (TobyAsE) [reply Userland/Libraries/LibGfx/BitmapMixer.cpp:43]: beautiful!

**Review by IdanHo — APPROVED** [review]
Thanks!

**Pull request merged**